docs/runs.md

```diff
@@ -0,0 +1,227 @@
+<!--
+---
+linkTitle: "Runs"
+weight: 2
+---
+-->
+
+# `Run`s
+
+- [Overview](#overview)
+- [Configuring a `Run`](#configuring-a-run)
+  - [Specifying the target Custom Task](#specifying-the-target-custom-task)
+  - [Specifying `Parameters`](#specifying-parameters)
+- [Monitoring execution status](#monitoring-execution-status)
+  - [Monitoring `Results`](#monitoring-results)
+- [Code examples](#code-examples)
+  - [Example `Run` with a referenced custom task](#example-run-with-a-referenced-custom-task)
+  - [Example `Run` with an unnamed custom task](#example-run-with-an-unnamed-custom-task)
+  - [Example of specifying parameters](#example-of-specifying-parameters)
+
+# Overview
+
+A `Run` allows you to instantiate and execute a [Custom
+Task](https://github.com/tektoncd/community/blob/master/teps/0002-custom-tasks.md),
+which can be implemented by a custom task controller running on-cluster. Custom
+Tasks can implement behavior that doesn't correspond directly to running a
+workload in a `Pod` on the cluster. For Pod-based on-cluster workloads, you
+should use a [`TaskRun`](taskruns.md).
+
+In order for a `Run` to actually execute, there must be a custom task
+controller running on the cluster that is responsible for watching and updating
+`Run`s which reference their type. If no such controller is running, `Run`s
+will have no `.status` value and no further action will be taken.
+
+`Run`s are an **_experimental alpha feature_** and should be expected to change
+in breaking ways or even be removed.
+
+## Configuring a `Run`
+
+A `Run` definition supports the following fields:
+
+- Required:
+  - [`apiVersion`][kubernetes-overview] - Specifies the API version, for example
+    `tekton.dev/v1beta1`.
+  - [`kind`][kubernetes-overview] - Identifies this resource object as a `Run` object.
+  - [`metadata`][kubernetes-overview] - Specifies the metadata that uniquely identifies the
+    `Run`, such as a `name`.
+  - [`spec`][kubernetes-overview] - Specifies the configuration for the `Run`.
+    - [`ref`](#specifying-the-target-custom-task) - Specifies the type and
+      (optionally) name of the custom task type to execute.
+- Optional:
+  - [`params`](#specifying-parameters) - Specifies the desired execution
+    parameters for the custom task.
+
+[kubernetes-overview]:
+  https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields
+
+### Specifying the target Custom Task
+
+To specify the custom task type you want to execute in your `Run`, use the
+`ref` field as shown below:
+
+```yaml
+spec:
+  ref:
+    apiVersion: example.dev/v1alpha1
+    kind: Example
+```
+
+This initiates the execution of a `Run` of a custom task of type `Example`, in
+the `example.dev` API group,  with the version `v1alpha1`.
+
+You can also specify the `name` and optional `namespace` (default is `default`)
```

Comment:
This is more of a design question than a documentation one - but do we want to allow for cross-namespace access of runs like this? It differs from how TaskRuns can access Tasks - and makes RBAC on these harder.

Reply:
Yeah good question, and I'm not sure. I haven't looked at it yet. AFAIK traditional Tasks can only be referenced from the Pipeline's namespace, or as a `ClusterTask`, so that's probably a model worth following until someone asks for more.

Reply:
+1 to following that same model.